src/lib/components/AccessRightField.js

```diff
@@ -111,13 +121,47 @@ export class AccessRightField extends Component {
   }
 }
 
-AccessRightField.propTypes = {
+class FormikAccessRightField extends Component {
+  render() {
+    // NOTE: This is a "cute" optimization.
+    //       In general, FastField only re-renders if
+    //       * formik slice associated with this.props.fieldPath changes
+    //         (i.e. `access` changes)
+    //       * props are ADDED or REMOVED to FastField
+    // So we add/remove a prop to FastField based on the presence of files.
+    // This way, FastField only renders when the things (access and hasFiles)
+    // it cares about change as it should be.
+    const change = this.props.hasFiles ? {change: true} : {}
+    return (
+      <FastField
+        name={this.props.fieldPath}
+        component={(formikProps) => (
+          <AccessRightFieldComponent formik={formikProps} {...this.props} />
+        )}
+        {...change}
+      />
+    );
+  }
+}
+
+FormikAccessRightField.propTypes = {
   fieldPath: PropTypes.string.isRequired,
   label: PropTypes.string.isRequired,
   labelIcon: PropTypes.string,
-  options: PropTypes.array,
+  hasFiles: PropTypes.bool,
 };
 
-AccessRightField.defaultProps = {
-  fieldPath: 'access.access_right',
+FormikAccessRightField.defaultProps = {
+  fieldPath: 'access',
 };
+
+
+const mapStateToProps = (state) => ({
+  hasFiles: Object.values(state.files.entries).length > 0,
```

Comment:
I think that now we have more places that this transformation occurs so maybe we should just store it in the redux state once a file action is triggered instead...